### PR TITLE
Do not allow adding Horizontal/Vertical containers inside of containers with the same type

### LINKED ISSUE #5082: Don't allow "same-type" linear container children creation
Nested linear containers are currently auto-merged, since it doesn't make any sense to have a, e.g., horizontal container directly inside another horizontal container. From a viewport perspective, this is strictly equivalent to a single horizontal container with all the children.

```text
horizontal
└ A
└ horizontal
   └ B
   └ C
```

same as:

```text
horizontal
└ A
└ B
└ C
```

This means that creating a Horizontal/Vertical container inside of a container of the same type, e.g. with the "Add SV/Container" modal, is a noop that is surprising to the user:

Export-1707301705100

==> The modal should disable "same-type-linear" container with a tooltip explaining why.

### PR DESCRIPTION
**Pull request by abey79**
### What

- Fixes #5082

*Note*: due to https://github.com/emilk/egui/issues/3997, the tooltip shows up immediately, which is slightly annoying.

image

image

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using newly built examples: [app.rerun.io](https://app.rerun.io/pr/5091/index.html)
  * Using examples from latest `main` build: [app.rerun.io](https://app.rerun.io/pr/5091/index.html?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [app.rerun.io](https://app.rerun.io/pr/5091/index.html?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](tests/python/release_checklist)!

- [PR Build Summary](https://build.rerun.io/pr/5091)
- [Docs preview](https://rerun.io/preview/57a22b0a289369057acc1000af69df20df9e3d23/docs) 
- [Examples preview](https://rerun.io/preview/57a22b0a289369057acc1000af69df20df9e3d23/examples) 
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

### CODE REVIEW

File: crates/re_viewport/src/add_space_view_or_container_modal.rs [hunk 1]
```diff
@@ -70,7 +71,32 @@ fn modal_ui(
...
+                    .tiles
+                    .get(blueprint_id_to_tile_id(&container_id))
+            })
+            .and_then(|tile| tile.container_kind());
+
+        // We disallow creating "linear" containers (horizontal/vertical) inside containers of the same kind, because
+        // it's not useful and is automatically simplified away.
+        let disabled = Some(kind) == target_container_kind
+            && (target_container_kind == Some(egui_tiles::ContainerKind::Horizontal)
+                || target_container_kind == Some(egui_tiles::ContainerKind::Vertical));
```
Comment (emilk) [inline crates/re_viewport/src/add_space_view_or_container_modal.rs:87]:
```suggestion
        let disabled = Some(kind) == target_container_kind
            && matches!(kind, egui_tiles::ContainerKind::Horizontal | egui_tiles::ContainerKind::Vertical);
```

File: crates/re_viewport/src/add_space_view_or_container_modal.rs [hunk 2]
```diff
@@ -70,7 +71,32 @@ fn modal_ui(
     ];
 
     for (title, subtitle, kind) in container_data {
-        if row_ui(ui, icon_for_container_kind(&kind), title, subtitle).clicked() {
+        let target_container_kind = target_container
```
Comment (emilk) [inline crates/re_viewport/src/add_space_view_or_container_modal.rs:74]: maybe `parent_container_kind` is more descriptive?

**Review by emilk — APPROVED** [review]

File: crates/re_viewport/src/add_space_view_or_container_modal.rs (hunk 2 above)
Comment (abey79) [reply crates/re_viewport/src/add_space_view_or_container_modal.rs:74]: We have "target_container" all over the place in that file, so I'd keep it as is for consistency.

File: crates/re_viewport/src/add_space_view_or_container_modal.rs (hunk 1 above)
Comment (abey79) [reply crates/re_viewport/src/add_space_view_or_container_modal.rs:87]: Yup, nice.

**Pull request merged**